src/pykeen/models/unimodal/node_piece.py

```diff
@@ -148,12 +155,15 @@ def __init__(
             )
 
         # always create representations for normal and inverse relations and padding
+        # TODO: always? or only if we need relation tokenization?
         relation_representations = embedding_specification.make(
             num_embeddings=2 * triples_factory.real_num_relations + 1,
         )
```

Comment:
relation_representation should be there in the model anyways (as the source of relation embeddings), but the `token_representation` sent to the `NodePieceRepresentation` constructor can vary:
* only one`relation_representation` - if we solely use `RelationTokenizer`
* only one `anchor_representation` (of shape `num_anchors + 1, emb_dim`, +1 for the padding) - if we solely use `AnchorTokenizer`
* both relation and anchor representations if we go full-mode with both anchor and relation tokenization

Reply:
For the last variant, we currently do not have a tokenizer implementation